Adds public and private character stats that can be used via events

Vassyli opened the pull request:
This PR updates dependencies and adds public and private character stats. Public character stats can be pulled for a "list of fighters", private ones for "current character stats", to be used during scene changes, for example.

Vassyli committed:
Introduction of CharacterStatTypes with event hooks.

Vassyli committed:
Adjusted error handling test for overblog/graphqlbundle 0.11

Vassyli committed:
Adds working test for publicStats

Vassyli committed:
Adds working test for privateStats accessed by anonymous

Vassyli committed:
Updated depdendencies.

Vassyli committed:
Added KERNEL_DIR

Vassyli committed:
Changed Character to extend from Fighter

Vassyli committed:
Implemented fields from Fighter Interface, changed test and added test for authenticated accessing of privateStats

Vassyli committed:
Adds tests for character stats

austenmc commented on app/config/config.yml:
```diff
@@ -51,8 +51,13 @@ swiftmailer:
 overblog_graphql:
     definitions:
-        internal_error_message: "An error occurred, please retry later or contact us!"
+        #internal_error_message: "An error occurred, please retry later or contact us!"
```
Can we not have comments?

austenmc commented on app/config/routing.yml:
```diff
@@ -19,5 +19,5 @@ overblog_graphql_endpoint:
-overblog_graphql_graphiql:
-    resource: "@OverblogGraphQLBundle/Resources/config/routing/graphiql.yml"
\ No newline at end of file
+#overblog_graphql_graphiql:
```
Do we really need comments here? We should just remove the code and keep it in the git history if we need it.

austenmc commented on src/AppBundle/GraphQL/Types/CharacterType.php:
```diff
@@ -50,4 +61,101 @@ public function getDisplayName(): string
+     * Returns public and private stats.
+     * @return array
+     */
+    public function getPrivateStats(): array
```
I dont like that getPrivateStats() returns both public and private. Either change it to getPublicAndPrivateStats or have it return only the private stats.

austenmc commented on src/AppBundle/GraphQL/Types/CharacterType.php:
```diff
@@ -50,4 +61,101 @@ public function getDisplayName(): string
+     */
+    private function returnBaseStats(): array
+    {
+        $stats = [
```
nit: remove newline

austenmc reviewed (changes requested):
In general this looks good, but Im not sure how useful the distinction between public and private stats are. Why not just have "stats" and then the system returns all the stats you are authorized to see?

Added some small nits and suggestions as well.

Vassyli commented:
Yeah, that's a good suggestion. I'll look into changing it to "stats", which gets rows from `getPublicStats()` and, if the character is authorized, `getPrivateStats()`.

Vassyli committed:
character fields privateStats and publicStats are now joined in stats

To ease access, statistics and status of characters are no longer available under different fields depending
if they are restricted or not. The two hooks, h/lotgd/crate-graphql/characterStats/private and /public still
exist and can be used accordingly for public information (like name, level, race) or private (like gold, gems).

Types extending BaseType can now call the method isGuarded to check whether or not access to the type is
restricted. This is used to only gather private stats if the user is actually authenticated to do so.

Vassyli committed:
Small code cleanup

Vassyli committed:
Added missing documentations

Vassyli commented:
Done.

austenmc commented:
Looks good. Let’s merge.
